src/UnixStamp.cpp

```diff
@@ -1,108 +1,114 @@
 #include "UnixStamp.hpp"
 
-UnixStamp::UnixStamp(time_t unix)
+UnixStamp::UnixStamp(unixstamp unix)
 {
     UnixStamp(unix, 0);
 }
 
-UnixStamp::UnixStamp(time_t unix, int8_t tz)
+UnixStamp::UnixStamp(unixstamp unix, int8_t tz)
 {
     this->unix = unix;
-    this->tz = tz;
     this->time = convertUnixToTime(unix, tz);
 }
 
-UnixStamp::UnixStamp(tm time)
+UnixStamp::UnixStamp(civil_time time)
 {
     UnixStamp(time, 0);
 }
 
-UnixStamp::UnixStamp(tm time, int8_t tz)
+UnixStamp::UnixStamp(civil_time time, int8_t tz)
 {
     this->time = time;
-    this->tz = tz;
-    this->unix = convertTimeToUnix(time, tz);
+    this->unix = convertTimeToUnix(time);
 }
 
-time_t UnixStamp::convertTimeToUnix(tm time, int8_t tz)
+unixstamp UnixStamp::convertTimeToUnix(civil_time time)
 {
-    time.tm_hour += tz;
-    return mktime(&time);
+    uint16_t year = time.year;
+    uint8_t month = time.mon;
+    year -= month <= 2;
+    uint8_t era = ((year >= 0 ? year : (year - YEARS_IN_ERA - 1)) / YEARS_IN_ERA);
+    uint16_t yearOfEra = year % YEARS_IN_ERA;
+    uint16_t dayOfYear = ((DAYS_AFTER_FIRST_MARCH * (month > 2 ? month - 3 : month + 9) + 2) / 5) + time.day - 1;
```

Comment:
I am a bit worried about this subtractions from unsigned values, but not 100% sure there is a problem here (because coercion 🙄). Would need to do some hands-on tests to be sure.

Reply:
Ah, no, it's all good here. The subtraction only happens conditionally, when it won't lead to underflow.

Reply:
You know, I'm worried too. But for another reason. A couple of tests for class instances were added, they are failing.